apps/comps/components/siwe/index.tsx

```diff
@@ -12,60 +14,64 @@ import {
   DropdownMenuTrigger,
 } from "@recallnet/ui2/components/shadcn/dropdown-menu";
 
-import { Identicon } from "@/components/Identicon";
-import { useLogout } from "@/hooks/useAuth";
-import { userAtom } from "@/state/atoms";
+import { useLogout } from "@/hooks";
+import { useProfile } from "@/hooks/useProfile";
 
 export const SIWEButton: React.FunctionComponent<
   React.ComponentProps<typeof Button>
 > = () => {
-  const [user, setUser] = useAtom(userAtom);
-  const { mutate: logout } = useLogout();
+  const router = useRouter();
+  const logout = useLogout();
+  const { data: user, isError, refetch } = useProfile();
 
-  const handleLogout = () => {
-    logout(undefined, {
-      onSuccess: () => {
-        setUser({ loggedIn: false, address: "" });
-      },
-    });
+  const handleLogout = async () => {
+    logout.mutate();
+    await refetch();
+  };
+
+  const handleMyAccount = () => {
+    router.push("/user/agents");
```

Comment:
when we merge https://github.com/recallnet/js-recall/pull/400 this link should be updated to `/profile`